error: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "provided_test_port2"]). You cannot derive ComponentDefinition in these cases, as you must resolve the ambiguity manually.
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Ambiguous port type: There are multiple fields with type TestPort (["required_test_port1", "required_test_port2"]). You cannot derive ComponentDefinition in these cases, as you must resolve the ambiguity manually.
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0407]: method `provided_ref` is not a member of trait `RequireRef`
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^ not a member of trait `RequireRef`
   |
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0407]: method `connect_to_required` is not a member of trait `RequireRef`
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^
   |          |
   |          not a member of trait `RequireRef`
   |          help: there is an associated function with a similar name: `connect_to_provided`
   |
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `DuplicatePortComponent: NetworkActor` is not satisfied
  --> $DIR/component_derive_tests.rs:12:10
   |
12 |     ctx: ComponentContext<Self>,
   |          ^^^^^^^^^^^^^^^^^^^^^^ the trait `NetworkActor` is not implemented for `DuplicatePortComponent`
   |
   = note: required because of the requirements on the impl of `kompact::prelude::Actor` for `DuplicatePortComponent`
   = note: required because of the requirements on the impl of `ActorRaw` for `DuplicatePortComponent`
   = note: required because of the requirements on the impl of `ComponentTraits` for `DuplicatePortComponent`
note: required by a bound in `kompact::component::ComponentContext`
  --> $DIR/context.rs:28:33
   |
28 | pub struct ComponentContext<CD: ComponentTraits> {
   |                                 ^^^^^^^^^^^^^^^ required by this bound in `kompact::component::ComponentContext`

error[E0277]: the trait bound `DuplicatePortComponent: ComponentLifecycle` is not satisfied
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^ the trait `ComponentLifecycle` is not implemented for `DuplicatePortComponent`
   |
note: required by a bound in `kompact::component::ComponentDefinition`
  --> $DIR/definition.rs:30:32
   |
30 | pub trait ComponentDefinition: DynamicComponentDefinition
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `kompact::component::ComponentDefinition`
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `DuplicatePortComponent: NetworkActor` is not satisfied
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^ the trait `NetworkActor` is not implemented for `DuplicatePortComponent`
   |
   = note: required because of the requirements on the impl of `kompact::prelude::Actor` for `DuplicatePortComponent`
   = note: required because of the requirements on the impl of `ActorRaw` for `DuplicatePortComponent`
note: required by a bound in `kompact::component::ComponentDefinition`
  --> $DIR/definition.rs:30:32
   |
30 | pub trait ComponentDefinition: DynamicComponentDefinition
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `kompact::component::ComponentDefinition`
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `DuplicatePortComponent: NetworkActor` is not satisfied
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^ the trait `NetworkActor` is not implemented for `DuplicatePortComponent`
   |
   = note: required because of the requirements on the impl of `kompact::prelude::Actor` for `DuplicatePortComponent`
   = note: required because of the requirements on the impl of `ActorRaw` for `DuplicatePortComponent`
   = note: required because of the requirements on the impl of `ComponentTraits` for `DuplicatePortComponent`
note: required by a bound in `kompact::component::Component`
  --> $DIR/actual_component.rs:50:26
   |
50 | pub struct Component<CD: ComponentTraits> {
   |                          ^^^^^^^^^^^^^^^ required by this bound in `kompact::component::Component`
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `DuplicatePortComponent: NetworkActor` is not satisfied
  --> $DIR/component_derive_tests.rs:10:10
   |
10 | #[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "pro...
   |          ^^^^^^^^^^^^^^^^^^^ the trait `NetworkActor` is not implemented for `DuplicatePortComponent`
   |
   = note: required because of the requirements on the impl of `kompact::prelude::Actor` for `DuplicatePortComponent`
   = note: required because of the requirements on the impl of `ActorRaw` for `DuplicatePortComponent`
   = note: required because of the requirements on the impl of `ComponentTraits` for `DuplicatePortComponent`
note: required by a bound in `kompact::component::ComponentContext`
  --> $DIR/context.rs:28:33
   |
28 | pub struct ComponentContext<CD: ComponentTraits> {
   |                                 ^^^^^^^^^^^^^^^ required by this bound in `kompact::component::ComponentContext`
   = note: this error originates in the derive macro `ComponentDefinition` (in Nightly builds, run with -Z macro-backtrace for more info)
